core/src/main/java/com/linecorp/armeria/common/util/CompositeException.java

```diff
@@ -142,25 +160,27 @@ public synchronized Throwable getCause() { // NOPMD
                         aggregateMessage.append("|-- ");
                         aggregateMessage.append(inner.getClass().getCanonicalName()).append(": ");
                         final String innerMessage = inner.getMessage();
+                        final String messagePadding = Strings.repeat("  ", depth + 2);
                         if (innerMessage != null && innerMessage.contains(separator)) {
                             aggregateMessage.append(separator);
                             for (String line : innerMessage.split(separator)) {
-                                for (int i = 0; i < depth + 2; i++) {
-                                    aggregateMessage.append("  ");
-                                }
-                                aggregateMessage.append(line).append(separator);
+                                aggregateMessage.append(messagePadding).append(line).append(separator);
                             }
                         } else {
                             aggregateMessage.append(innerMessage);
                             aggregateMessage.append(separator);
                         }
 
-                        for (int i = 0; i < depth + 2; i++) {
-                            aggregateMessage.append("  ");
-                        }
                         final StackTraceElement[] st = inner.getStackTrace();
                         if (st.length > 0) {
-                            aggregateMessage.append("at ").append(st[0]).append(separator);
+                            final boolean isVerboseException =
+                                    verboseExceptionSampler.isSampled(inner.getClass());
+                            final int maxStackTraceSize = isVerboseException ? st.length : Math.min(
+                                                          DEFAULT_MAX_NUM_STACK_TRACES, st.length);
+                            for (int i = 0; i < maxStackTraceSize; i++) {
+                                aggregateMessage.append(messagePadding).append("at ").append(st[i]).append(
+                                        separator);
```

Comment:
nit: For readability.
```suggestion
                                aggregateMessage.append(messagePadding).append("at ")
                                                .append(st[i]).append(separator);
```